File: resources/sample.clj
;; Clojure Sample for Something Theme

(ns something.sample
  "A sample namespace to demonstrate Clojure syntax highlighting."
  (:require [clojure.string :as str]))



(defonce ^:private app-state (atom {:count 0}))

(defn- my-private-function [arg]
  (str "Hello, " arg "!"))

(defmacro unless [pred a b]
  `(if (not ~pred) ~a ~b))

(defn public-function
  "This is a public function with a docstring."
  [name]
  (let [greeting (my-private-function name)
        counter (swap! app-state update :count inc)]
    (println greeting)
    (println (str/join " " ["Counter:" (:count counter)]))
    (unless (= name "World")
      (println "You are not the world.")
      nil)))

(public-function "Something")

File: resources/sample.py
# Python Sample for Something Theme
from typing import List

class Greeter:
    """A sample class to demonstrate Python syntax."""

    # A class attribute
    PI = 3.14159

    def __init__(self, name: str):
        self.name = name

    def greet(self, loud: bool = False):
        """Greets the person."""
        greeting = f"Hello, {self.name}!"
        if loud:
            print(greeting.upper())
        else:
            print(greeting)

if __name__ == "__main__":
    greeter = Greeter("Something")
    greeter.greet()
    greeter.greet(loud=True)